drivers/power_domain/power_domain_tisci.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2025 Texas Instruments
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/firmware/tisci/tisci.h>
+#include <zephyr/drivers/power/ti,sci_pm_domain.h>
```

Comment:
The enum in this file no longer need to be public, the header can be removed and the enums defined in this source file :)